### DISCUSSION TITLE
升级到NET8.0后，在HubConnection的On方法中，调用Table的QueryAsync()方法，无法显示数据。

### DISCUSSION THREAD

**Discussion by yguang81** [dotnetcore/BootstrapBlazor#3769] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

之前是在NET6.0环境中，可以正常显示数据，升级到NET8.0后，无法正常显示数据。
HubConnection的On方法中调用Table的QueryAsync()方法的代码如下  
``` C#
hubConnection.On<string, List<DeviceTagInfo>>("ReceiveDeviceVariablesNotify", async (deviceName, deviceTagInfos) =>
 {
     if (deviceName == DeviceName)
     {
         var encodedMsg = $"Device KeyPairs: {deviceTagInfos}";
         DeviceTagInfos = deviceTagInfos;
         await TableElement.QueryAsync();
     }
 });
```

Table的OnQueryAsync事件的实现如下  
``` C#
private async Task<QueryData<DeviceTagInfo>> OnQueryAsync(QueryPageOptions options)
{

    List<DeviceTagInfo> items = DeviceTagInfos;
    // 设置记录总数
    var total = items.Count();

    // 内存分页
    items = items.Skip((options.PageIndex - 1) * options.PageItems).Take(options.PageItems).ToList();

    return new QueryData<DeviceTagInfo>() { Items = items, TotalCount = total };

}
```

执行完OnQueryAsync所有方法后，VS自动进入以下部分进行报错。
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/44687803/98bc5218-365d-4d25-916f-49ab10438af0)

通过测试，把BootstrapBlazor的Table组件中的QueryAsync方法中 ，**StateHasChanged** 换成异步调用就可以了，**await InvokeAsync(StateHasChanged);**  
``` C#
    private async Task QueryAsync(bool shouldRender, int? pageIndex = null)
    {
        if (ScrollMode == ScrollMode.Virtual && VirtualizeElement != null)
        {
            await VirtualizeElement.RefreshDataAsync();
        }
        else
        {
            await InternalToggleLoading(true);
            if (pageIndex.HasValue)
            {
                PageIndex = pageIndex.Value;
            }
            await QueryData();
            await InternalToggleLoading(false);
        }

        if (shouldRender)
        {
            //StateHasChanged
            //此部分换成异步调用
            await InvokeAsync(StateHasChanged);
        }
    }
```

**Comment by bb-auto** [comment]
@yguang81 Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@yguang81 项目右上角帮忙点一下 star 谢谢啦。请尽快提供复现示例

**Comment by yguang81** [comment]
我在gitee的仓库中创建一个Demo程序。地址如下：
[https://gitee.com/yguang81/bootstrap-blazor-test.git](https://gitee.com/yguang81/bootstrap-blazor-test.git)

仓库中，分别有NET6与NET8的环境的Dome. 在仓库中有放置一个GIF动画，演示了我复现的问题点.
[BootstrapBlazor测试1.gif](https://gitee.com/yguang81/bootstrap-blazor-test/blob/master/BootstrapBlazor%E6%B5%8B%E8%AF%951.gif)

**Reply by ArgoZhang** [reply, marked as answer]
@yguang81 感谢提交详细的测试步骤与示例

`TableDemoVariables.razor` 文件 56 行更改如下即可
```
await InvokeAsync(() => TableElement.QueryAsync());
```

请帮忙点一下下方的 `make as answer` 按钮标记此答案，方便其他人解决类似问题
